tests/project/test_project.py

```diff
@@ -39,6 +39,21 @@ def test_data(self):
         self.assertEqual(1, len(project2.data))
         self.assertEqual(self.project.data.foo, project2.data.foo)
 
+    def test_iterjobs(self):
+
+        try:
+            for j in self.project.iter_jobs():
+                pass
+        except:
+            self.fail("Iterating over empty project should not raise exception.")
```

Comment:
Since a test fails if an exception is raised and this way we do not actually see the exception any more I would prefer
```suggestion
        except Exception as e:
            raise AssertionError("Iterating over empty project should not raise exception.") from e
```

Reply:
When I check the test output at 92d87d4 the key error also pops up in the output.  In this case I'd prefer calling the `fail` method, because that makes it clear as a test failure not an exception.  I know raising `AssertionError` is handled by `unittest` as a test failure, so I won't insist, it's a just a style choice.

Reply:
I agree, its just about style - and as long as the `self.fail` also shows/includes the raised Exception from the code I am happy with either solution.